src/librustc/hir/def.rs

```diff
@@ -277,140 +334,85 @@ impl NonMacroAttrKind {
     }
 }
 
-impl<Id> Def<Id> {
+impl<Id> Res<Id> {
     /// Return the `DefId` of this `Def` if it has an id, else panic.
     pub fn def_id(&self) -> DefId
     where
         Id: Debug,
     {
         self.opt_def_id().unwrap_or_else(|| {
-            bug!("attempted .def_id() on invalid def: {:?}", self)
+            bug!("attempted .def_id() on invalid res: {:?}", self)
         })
     }
 
-    /// Return `Some(..)` with the `DefId` of this `Def` if it has a id, else `None`.
+    /// Return `Some(..)` with the `DefId` of this `Res` if it has a id, else `None`.
```

Comment:
```suggestion
    /// Return `Some(..)` with the `DefId` of this `Res` if it has an id, else `None`.
```

Reply:
Preexisting, I'd rather not do drive-by changes like that in something this big.

Reply:
🤷‍♂ it's a trivial change that I'm sure will be forgotten otherwise.

Reply:
Feel free to open an issue or another PR.